### ISSUE TITLE
Does Flye trim adaptors?

### ISSUE THREAD

**Issue by vviala** [fenderglass/ABruijn#100]
Greatings. Does Flye trim nanopore adaptors? When you say it handles raw reads, you mean fastq directly from basecalling? Thanks in advance.

**Comment by mikolmogorov** [comment]
Hi,

No, Flye does not explicitly cut adapters. But I don't think I saw adapters in nanopore WGS reads before. Is this a multiplexed run? Flye uses basecalled fastqs as input (as you said), but we assume that reads only contain genomic sequence.

**Comment by vviala** [comment]
O Mikhail, kind of you to answer.
Thanks.
Yes. Nanopore data have sequencing adaptors ( other than barcodes for multiplexing. ) As far as I konw, the Guppy basecaller does not trim that. Thanks for the answer. Your asembler looks great.
As we are discussing here, if you don't mind :
I am assembling BAC clones. Circular DNA with aprox 200kb. Larger than common plasmids. Does your code works for circular assemblies other than short plasmids? O tried to use the --plasmid command. Flye assembled correctly the BAC but linear. André the plasmid folder did not contain circular sequence.

Thanks again


Em 16 de abr de 2019 17:44, Mikhail Kolmogorov <notifications@github.com> escreveu:

Hi,

No, Flye does not explicitly cut adapters. But I don't think I saw adapters in nanopore WGS reads before. Is this a multiplexed run? Flye uses basecalled fastqs as input (as you said), but we assume that reads only contain genomic sequence.

**Comment by mikolmogorov** [comment]
Thank you! Yes, it seems that you can cut adapters using Porechop (https://github.com/rrwick/Porechop). But as I understand the adapters are rather short. Flye does account for inaccurate read ends, so assembling reads with short adapters at the ends should be smooth.

Flye should be able to assemble circular sequences, yes. The `--plasmid` mode enables extra algorithms that focus on short circular sequences (that could otherwise be missed), but the output is combined with the rest of the assembled contigs.

**Comment by vviala** [comment]
Nice. Can I track down which contig was assembled as a circular sequence? Maybe using the "edge" data? Because otherwise I have to check for ends overlapping manually by self-plot but it is quite craft work.
Congratulations for your workshop
Best

Em 16 de abr de 2019 19:26, Mikhail Kolmogorov <notifications@github.com> escreveu:

Thank you! Yes, it seems that you can cut adapters using Porechop (https://github.com/rrwick/Porechop). But as I understand the adapters are rather short. Flye does account for inaccurate read ends, so assembling reads with short adapters at the ends should be smooth.

Flye should be able to assemble circular sequences, yes. The --plasmid mode enables extra algorithms that focus on short circular sequences (that could otherwise be missed), but the output is combined with the rest of the assembled contigs.

**Comment by mikolmogorov** [comment]
Yes, you can check the `assembly_info.txt` file - it contains various extra information about contigs, including the tags for circular sequences.

**Comment by mikolmogorov** [comment]
Closing for now - feel free to reopen if you have more questions.

**Comment by elina2410** [comment]
Hello! I trim barcodes in nanopore reads using Porechop before assembling.
I also wonder if I should run Nanofilt on ONT reads (eg -q 15) or just leave reads as they are?
Flye expected error rate 10-15% with --nanopore-raw flag. I had E. coli strain with 2 plasmids . And I tried both (1) quality trimmed (-q15 Nanofilt) and (2) non-trimmed reads for assembling it with flag --nanopore-raw. I obtained different results in contiguity:
(1) 3 contigs, non of them circular
(2) 3 circular contigs.   
I know that in flye.log I can find info about error rate, and I found these regarding to non-trimmed by Nanofilt  
[2023-03-15 07:22:42] root: INFO: Alignment error rate: 0.089280
[2023-03-15 07:30:18] root: INFO: Alignment error rate: 0.049608
Based on this can I assume that my error rate is even lower than Flye expects and there is no need for quality trimming? Especially when I lost the contiguity of contigs? For trimmed reads: Alignment error rate: 0.048239,  Alignment error rate: 0.02613
On the other hand, I performed also hybrid assembly (illumina+ONT reads) and performed cgMLST analysis for those 3 assemblies. Assuming that the hybird assembly is the most accurate, the flye assembly (1) with trimmed reads is more similar to it (113 allele differences) than (2) non-trimmed reads assembly (197 allele differences). 
So should I quality-trim ONT reads or not? What is recommended for Flye?
I would like to achieve a balance between the accuracy of the assembly and the contiguity of contigs.
